### ISSUE TITLE
A question about the dataset

### ISSUE THREAD

**Issue by jujingliuzy** [windrise/3DGR-CAR#8]
Dear author: 
    Congratulations, this is a great work！
    But I still have a small question. I want to know how to generate the **_volume.pt_**  file from the nii.gz file in the ImageCAS dataset,  which is an input to **_ct_projector_train.forward_project_**() .
    I would appreciate it if you could reply soon.

Best wishes,
Zhenyu Liu

**Comment by windrise** [comment]
The relevant preprocessing code is as follows：
`def crop_to_standard(scan, scale):
    z, y, x = scan.shape
    if z >= scale:
        ret_scan = scan[z-scale:z, :, :]
        # ret_scan = scan
    else:
        temp1 = np.zeros(((scale-z)//2, y, x))
        temp2 = np.zeros(((scale-z)-(scale-z)//2, y, x))
        ret_scan = np.concatenate((temp1, scan, temp2), axis=0)
    z, y, x = ret_scan.shape
    if y >= scale:
        ret_scan = ret_scan[:, (y-scale)//2:(y+scale)//2, :]
    else:
        temp1 = np.zeros((z, (scale-y)//2, x))
        temp2 = np.zeros((z, (scale-y)-(scale-y)//2, x))
        ret_scan = np.concatenate((temp1, ret_scan, temp2), axis=1)
    z, y, x = ret_scan.shape
    if x >= scale:
        ret_scan = ret_scan[:, :, (x-scale)//2:(x+scale)//2]
    else:
        temp1 = np.zeros((z, y, (scale-x)//2))
        temp2 = np.zeros((z, y, (scale-x)-(scale-x)//2))
        ret_scan = np.concatenate((temp1, ret_scan, temp2), axis=2)
    return ret_scan
def resample_to_standard(scan, scale):
    z, y, x = scan.shape
    z_scale, y_scale, x_scale = scale / z, scale / y, scale / x
    resampled_scan = zoom(scan, (z_scale, y_scale, x_scale), order=1, mode='nearest')
    return resampled_scan 

itkimage = sitk.ReadImage(image_file)
image_mha = sitk.GetArrayFromImage(itkimage)
image_mha = crop_to_standard(image_mha, scale=320)
image_mha = resample_to_standard(image_mha, scale=scale_size)
image_mha = torch.tensor(image_mha, dtype=torch.float32)[None, ...]

image_size = [scale_size, scale_size, scale_size]
proj_size = [scale_size, scale_size, scale_size]

image_mha = image_mha.cuda()
num_proj=4
ct_projector = ConeBeam3DProjector(image_size, proj_size, num_proj)
projs = ct_projector.forward_project(image_mha.transpose(1, 4).squeeze(1))  # [bs, x, y, z] -> [bs, n, h, w]
print("projs shape: ",projs.shape)
`

**Comment by jujingliuzy** [comment]
Thanks for your code.
However, I have a question: if we apply *crop_to_standard* and then
*resample_to_standard*, wouldn't this crop out part of the region of
interest, making it unrecoverable after interpolation? So I want to know if
this is a common practice?

Xueming Fu ***@***.***> 于2025年6月27日周五 19:08写道：

**Comment by windrise** [comment]
The purpose of performing preprocessing in this way is to ensure that the coronary artery accounts for the main part of the image (the pixel part of the coronary artery will not be very small)
